src/messages/Lifecycle.ts

```diff
@@ -46,6 +46,14 @@ export interface EnvData {
   custom?: any;
 }
 
+/**
+ * Extended Environmental data passed to client with some private data.
+ */
+export interface EnvDataExt extends EnvData {
```

Comment:
I am not a fan of names like this (e.g Extended or Ext).  It doesn't give any meaning to the actual class and leads to people just dumping things into it because it has a generic name/purpose.

Reply:
How about something like `BooststrapData` or `SetupData`?

Reply:
I am pretty bad at naming variables. The purpose of having two kind of `EnvData` is that `EnvData` is the data exposed to client app and `ExtEnvData` contains `EndData` and some other private properties. Maybe change `ExtEnvData` to `RawEnvData`?

Reply:
> How about something like `BooststrapData` or `SetupData`?

Sounds great!